url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/news/utility/Stemmer.java

```diff
@@ -0,0 +1,562 @@
+			case 'e':
+			case 'i': 
+			case 'o': 
+			case 'u': 
+				return false;
+			case 'y': 
+				return (i==0) ? true : !cons(i-1);
+			default: return true;
+		}
+	}
+	
+	/* m() measures the number of consonant sequences between 0 and j. if c is
+	      a consonant sequence and v a vowel sequence, and <..> indicates arbitrary
+	      presence,
+	
+	         <c><v>       gives 0
+	         <c>vc<v>     gives 1
+	         <c>vcvc<v>   gives 2
+	         <c>vcvcvc<v> gives 3
+	         ....
+	 */
+	private final int m() {  
+		int n = 0;
+		int i = 0;
+		while(true) {  
+			if (i > j) 
+				return n;
+			if (! cons(i)) 
+				break; 
+			i++;
+		}
+		i++;
+		while(true) {  
+			while(true) {  
+				if (i > j) 
+					return n;
+				if (cons(i)) 
+					break;
+				i++;
+			}
+			i++;
+			n++;
+			while(true) {  
+				if (i > j) 
+					return n;
+				if (! cons(i)) 
+					break;
+				i++;
+			}
+			i++;
+		}
+	}
+	
+	/* vowelinstem() is true <=> 0,...j contains a vowel */
+	private final boolean vowelinstem() {  
+		int i; 
+		for (i = 0; i <= j; i++) 
+			if (! cons(i)) 
+				return true;
+		return false;
+	}
+	
+	/* doublec(j) is true <=> j,(j-1) contain a double consonant. */
+	private final boolean doublec(int j) {  
+		if (j < 1) 
+			return false;
+		if (b[j] != b[j-1]) 
+			return false;
+		return cons(j);
+	}
+	
+	/* cvc(i) is true <=> i-2,i-1,i has the form consonant - vowel - consonant
+	      and also if the second c is not w,x or y. this is used when trying to
+	      restore an e at the end of a short word. e.g.
+	
+	         cav(e), lov(e), hop(e), crim(e), but
+	         snow, box, tray.
+	
+	 */
+	private final boolean cvc(int i) {  
+		if (i < 2 || !cons(i) || cons(i-1) || !cons(i-2)) 
+			return false;
+		int ch = b[i];
+		if (ch == 'w' || ch == 'x' || ch == 'y') 
+			return false;
+		return true;
+	}
+	
+	private final boolean ends(String s)
+	{  
+		int l = s.length();
+		int o = k-l+1;
+		if (o < 0) 
+			return false;
+		for (int i = 0; i < l; i++) 
+			if (b[o+i] != s.charAt(i)) 
+				return false;
+		j = k-l;
+		return true;
+	}
+	
+	/* setto(s) sets (j+1),...k to the characters in the string s, readjusting
+	      k. */
+	
+	private final void setto(String s) {  
+		int l = s.length();
+		int o = j+1;
+		for (int i = 0; i < l; i++) 
+			b[o+i] = s.charAt(i);
+		k = j+l;
+	}
+	
+	/* r(s) is used further down. */
+	
+	private final void r(String s) { 
+		if(m() > 0)
+			setto(s); 
+	}
+	
+	/* step1() gets rid of plurals and -ed or -ing. e.g.
+	
+	          caresses  ->  caress
+	          ponies    ->  poni
+	          ties      ->  ti
+	          caress    ->  caress
+	          cats      ->  cat
+	
+	          feed      ->  feed
+	          agreed    ->  agree
+	          disabled  ->  disable
+	
+	          matting   ->  mat
+	          mating    ->  mate
+	          meeting   ->  meet
+	          milling   ->  mill
+	          messing   ->  mess
+	
+	          meetings  ->  meet
+	
+	 */
+	private final void step1() {  
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.